pkg/virt-operator/install-strategy/strategy.go

```diff
@@ -201,8 +209,14 @@ func GenerateCurrentInstallStrategy(config *operatorutil.KubeVirtDeploymentConfi
 	rbaclist = append(rbaclist, rbac.GetAllHandler(config.GetNamespace())...)
 
 	monitorNamespace := config.GetMonitorNamespace()
-	monitorServiceAccount := config.GetMonitorServiceAccount()
-	rbaclist = append(rbaclist, rbac.GetAllServiceMonitor(config.GetNamespace(), monitorNamespace, monitorServiceAccount)...)
+	if addMonitorServiceResources {
+		// TODO: we should check that monitor SA exists in the monitor namespace
+		monitorServiceAccount := config.GetMonitorServiceAccount()
+		rbaclist = append(rbaclist, rbac.GetAllServiceMonitor(config.GetNamespace(), monitorNamespace, monitorServiceAccount)...)
+		strategy.serviceMonitors = append(strategy.serviceMonitors, components.NewServiceMonitorCR(config.GetNamespace(), monitorNamespace, true))
+	} else {
+		glog.Errorf("failed to create service monitor resources because namespace %s does not exist", monitorNamespace)
```

Comment:
This should be an error?

if we are running on k8s we don't have the MonitorService right?

Reply:
Nope we only want to publish the error without interrupting the operator flow

Reply:
Hm, better `Warningf`.